src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -5,23 +5,91 @@ namespace Microsoft.Azure.ServiceBus.Core
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
     using System.Threading.Tasks;
+    using Microsoft.Azure.Amqp;
+    using Microsoft.Azure.Amqp.Encoding;
+    using Microsoft.Azure.Amqp.Framing;
+    using Microsoft.Azure.ServiceBus.Amqp;
+    using Microsoft.Azure.ServiceBus.Primitives;
 
-    internal abstract class MessageSender : ClientEntity, IMessageSender
+    public sealed class MessageSender : ClientEntity, IMessageSender
     {
-        [System.Diagnostics.CodeAnalysis.SuppressMessage(
-             "StyleCop.CSharp.ReadabilityRules",
-             "SA1126:PrefixCallsCorrectly",
-             Justification = "This is not a method call, but a type.")]
-        protected MessageSender(TimeSpan operationTimeout, RetryPolicy retryPolicy)
+        int deliveryCount;
+        readonly bool ownsConnection;
+
+        public MessageSender(ServiceBusConnectionStringBuilder connectionStringBuilder, RetryPolicy retryPolicy = null)
+            : this(connectionStringBuilder.GetNamespaceConnectionString(), connectionStringBuilder.EntityPath, retryPolicy)
+        {
+        }
+
+        public MessageSender(string connectionString, string entityPath, RetryPolicy retryPolicy = null)
+            : this(entityPath, new ServiceBusNamespaceConnection(connectionString), retryPolicy ?? RetryPolicy.Default)
+        {
+            if (string.IsNullOrWhiteSpace(connectionString))
+            {
+                throw Fx.Exception.ArgumentNullOrWhiteSpace(connectionString);
+            }
+            if (string.IsNullOrWhiteSpace(entityPath))
+            {
+                throw Fx.Exception.ArgumentNullOrWhiteSpace(entityPath);
+            }
+
+            this.ownsConnection = true;
+        }
+
+        public MessageSender(string entityPath, ServiceBusConnection serviceBusConnection, RetryPolicy retryPolicy = null)
+            : base(nameof(MessageSender) + StringUtility.GetRandomString(), retryPolicy ?? RetryPolicy.Default)
+        {
+            this.OperationTimeout = serviceBusConnection.OperationTimeout;
```

Comment:
>his.OperationTimeout = serviceBusConnection.OperationTimeout; [](start = 13, length = 61)

like MessageReceiver, have only one constructor do entire initialization. ALl other constructors should just drive to that one particular overload